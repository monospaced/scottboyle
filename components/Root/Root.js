import React, {Component} from 'react';
import '../../assets/favicon.ico';
import '../../assets/.htaccess';
import {version} from '../../package.json';

class Root extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {app, meta} = this.props;
    return (
      <html lang="en-gb">
        <head>
          <meta charSet="utf-8" />
          {meta}
          <meta name="viewport" content="width=device-width" />
          <link rel="stylesheet" href={`/styles.css?v=${version}`} />
        </head>
        <body>
          <div
            id="app"
            className="App"
            dangerouslySetInnerHTML={app}
          />
          <script src={`/bundle.js?v=${version}`}></script>
        </body>
      </html>
    );
  }
}

module.exports = Root;
