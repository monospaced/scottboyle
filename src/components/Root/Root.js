import React from "react";

import "../../assets/.htaccess";
import "./Root.css";
import Head from "../Head/Head";
import { version } from "../../../package.json";

const Root = ({ meta, router }) => {
  return (
    <html lang="en-gb">
      <Head meta={meta} />
      <body className="no-js">
        <script
          dangerouslySetInnerHTML={{ __html: `document.body.className = ""` }}
        />
        <div id="router" dangerouslySetInnerHTML={router} />
        <script src={`/bundle.js?v=${version}`} />
      </body>
    </html>
  );
};

export default Root;
