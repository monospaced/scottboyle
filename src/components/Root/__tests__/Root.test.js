import React from "react";
import { shallow } from "enzyme";

import Root from "../Root.js";

jest.mock("../../../../package.json", () => ({ version: "0.0.0" }));

describe("Root component", () => {
  it("should render correctly", () => {
    const component = shallow(<Root meta={[]} router={<div class="App" />} />);
    expect(component).toMatchSnapshot();
  });
});
