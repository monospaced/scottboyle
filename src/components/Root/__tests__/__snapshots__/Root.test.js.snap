// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should render correctly 1`] = `
<html
  lang="en-gb"
>
  <Head
    meta={Array []}
  />
  <body
    className="no-js"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "document.body.className = \\"\\"",
        }
      }
    />
    <div
      dangerouslySetInnerHTML={
        <div
          class="App"
        />
      }
      id="router"
    />
    <script
      src="/bundle.js?v=0.0.0"
    />
  </body>
</html>
`;
